packages/mcp/src/client/configuration.ts

```diff
@@ -56,22 +58,82 @@ To fix this you have three different options:
 3. If you only need one instance of MCPClient in your app, refactor your code so it's only created one time (ex. move it out of a loop into a higher scope code block)
 `);
       }
-      this.resources = existingInstance.resources; // Reuse resources object
       return existingInstance;
     }
 
     mcpClientInstances.set(this.id, this);
     this.addToInstanceCache();
-    this.resources = new ResourceClientActions({
-      getServerConfigs: () => this.serverConfigs,
-      getConnectedClient: (name, config) => this.getConnectedClient(name, config),
-      getConnectedClientForServer: (serverName) => this.getConnectedClientForServer(serverName),
-      addToInstanceCache: () => this.addToInstanceCache(),
-      getLogger: () => this.logger,
-    });
     return this;
   }
 
+  private async _getResourceActions(serverName: string): Promise<ResourceClientActions> {
+    if (this.resourceActionsCache.has(serverName)) {
+      return this.resourceActionsCache.get(serverName)!;
+    }
+
+    const serverConfig = this.serverConfigs[serverName];
+    if (!serverConfig) {
+      throw new Error(`Server configuration not found for name: ${serverName}`);
+    }
+
+    const internalClient = await this.getConnectedClientForServer(serverName);
+    
+    const actions = new ResourceClientActions({
+      client: internalClient,
+      logger: this.logger,
+    });
```

Comment:
I think this isn't the right approach, ResourceClientActions should be added to each individual client when the client is instantiated.

MCPClient -> handle multiple clients
internal client -> handle each kind of logic for a single client connection (including resources)

Then we don't need an in memory cache because each individual client is one already

Reply:
you should be able to do `internalClient.resources.list()` without all this extra indirection